lambdas/src/apis/profiles/controllers/profiles.ts

```diff
@@ -11,6 +11,45 @@ import { WearablesOwnership } from '../WearablesOwnership'
 
 const LOGGER = log4js.getLogger('profiles')
 
+function setCacheHeaders(res: Response, profiles: ProfileMetadata[]) {
+  if (profiles.length === 0) return
+
+  let maxTimestamp = profiles[0]?.timestamp
+  for (let i = 1; i < profiles.length; i++) {
+    maxTimestamp = Math.max(maxTimestamp, profiles[i].timestamp)
+  }
+
+  if (maxTimestamp) {
+    res.header('Last-Modified', new Date(maxTimestamp).toUTCString())
+    res.header('Cache-Control', 'public, max-age=0,s-maxage=0')
+  }
+}
+
+function getIfModifiedSinceTimestamp(req: Request): number | undefined {
+  const headerValue = req.header('If-Modified-Since')
+  if (!headerValue) return
+  try {
+    const timestamp = Date.parse(headerValue)
+    return isNaN(timestamp) ? undefined : timestamp
+  } catch (e) {
+    LOGGER.warn('Received an invalid header for IfModifiedSince ', headerValue)
+  }
+}
+
+function sendProfilesResponse(res: Response, profiles: ProfileMetadata[] | undefined, singleProfile: boolean = false) {
+  if (profiles) {
+    setCacheHeaders(res, profiles)
+    if (singleProfile) {
+      const returnProfile: ProfileMetadata = profiles[0] ?? { avatars: [], timestamp: 0 }
+      res.send(returnProfile)
+    } else {
+      res.send(profiles)
+    }
+  } else {
+    res.status(304).send()
```

Comment:
Please add a comment here explaining this case

Reply:
Oh here it was. this is what I was asking in the previous comment